source/common/buffer/watermark_buffer.cc

```diff
@@ -70,9 +71,19 @@ Api::IoCallUint64Result WatermarkBuffer::write(Network::IoHandle& io_handle) {
 
 void WatermarkBuffer::setWatermarks(uint32_t low_watermark, uint32_t high_watermark) {
   ASSERT(low_watermark < high_watermark || (high_watermark == 0 && low_watermark == 0));
+  uint32_t overflow_watermark_multiplier =
+      Runtime::getInteger("envoy.buffer.overflow_multiplier", 2);
```

Comment:
About 2x the high-watermark in an output buffer is expected due to how read limits interacts with the high watermark according to this comment.  We know H2 flow control windows may result in output buffers growing far beyond the configured high watermark.  Should we use a higher threshold for overflow watermark?  I guess 2x is fine as long as we don't take any drastic actions when overflow is hit.

https://github.com/envoyproxy/envoy/blob/1aed8d28c29fefeaa531f50ea3e1ff6d41d4fae3/source/common/network/connection_impl.cc#L451

Reply:
That's a very good point!
It is worth noting that the overflow threshold may be specific to a buffer functionality.
If this is the case, then a different design is needed.
For now I've kept it at 2 (as you've said, there are no drastic actions that are being taken at the moment), and if we see that it should be increased (rather than altering some other design), then we will increase this.

Reply:
You may want to default to 0 instead of 2 since this is a work-in-progress.

Reply:
Done